src/autopas/tuning/tuningStrategy/decisionTreeTuning/DecisionTreeTuning.cpp

```diff
@@ -0,0 +1,168 @@
+/**
+ * @file DecisionTreeTuning.cpp
+ * @author Abdulkadir Pazar
+ * @date 20.06.24
+ */
+
+#include "DecisionTreeTuning.h"
+
+#include <Python.h>
+
+#include <iostream>
+#include <sstream>
+
+namespace autopas {
+
+// Constructor
+DecisionTreeTuning::DecisionTreeTuning(const std::set<Configuration> &searchSpace, const std::string &modelFileName)
+    : _configurations(searchSpace), _modelFileName(modelFileName), _pFunc(nullptr) {
+  // Initialize Python Interpreter
+  Py_Initialize();
+  loadScript();
+}
+
+// Destructor
+DecisionTreeTuning::~DecisionTreeTuning() {
+  // Finalize Python Interpreter
+  if (_pFunc) {
+    Py_DECREF(_pFunc);  // Clean up the reference to _pFunc
+  }
+  Py_Finalize();
+}
+
+void DecisionTreeTuning::loadScript() {
+  // Add the current directory to the Python path
+  PyObject *sysPath = PySys_GetObject((char *)"path");
+  PyObject *cwd = PyUnicode_FromString(".");
+  PyList_Append(sysPath, cwd);
+  Py_DECREF(cwd);
+
+  // Load the fixed Python script predict.py
+  PyObject *pName = PyUnicode_DecodeFSDefault("predict");
+  if (!pName) {
+    throw std::runtime_error("Failed to convert script name to Python string.");
```

Comment:
Use AutoPas' exception handler as it provides some useful features like specifying rank ID and using `spdlog`.
```suggestion
    utils::ExceptionHandler::exception("Failed to convert script name to Python string.");
```
You will have to add `#include "autopas/utils/ExceptionHandler.h"`